src/test/java/nextstep/qna/domain/AnswerTest.java

```diff
@@ -1,8 +1,49 @@
 package nextstep.qna.domain;
 
-import nextstep.users.domain.NsUserTest;
+import nextstep.qna.CannotDeleteException;
+import nextstep.users.domain.NsUser;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.time.LocalDateTime;
+
+import static nextstep.fixture.AnswerFixture.*;
+import static nextstep.fixture.NsUserFixture.*;
+import static nextstep.fixture.QuestionFixture.*;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
 
 public class AnswerTest {
-    public static final Answer A1 = new Answer(NsUserTest.JAVAJIGI, QuestionTest.Q1, "Answers Contents1");
-    public static final Answer A2 = new Answer(NsUserTest.SANJIGI, QuestionTest.Q1, "Answers Contents2");
+
+    private Answer answer;
+
+    private NsUser loginUser;
+
+    @BeforeEach
+    void setUp() {
+        answer = new Answer(A1.getId(), A1.getWriter(), Q1, A1.getContents());
+        loginUser = A1.getWriter();
+    }
+
+    @DisplayName("다른 사람이 쓴 답변이 있다면 예외가 발생한다..")
+    @Test
+    void if_user_is_not_same_as_login_user_then_can_not_delete_question() {
+        assertThatThrownBy(() -> answer.delete(SANJIGI))
+                .isInstanceOf(CannotDeleteException.class);
+    }
+
+    @DisplayName("질문 데이터는 삭제 후 soft delete 처리가 된다.")
+    @Test
+    void set_delete_true_when_answer_was_deleted() {
+        DeleteHistory expected = new DeleteHistory(ContentType.ANSWER,
+                                                   A1.getId(),
+                                                   A1.getWriter(),
+                                                   LocalDateTime.now());
+
+        DeleteHistory actual = answer.delete(loginUser);
+
+        assertThat(answer.isDeleted()).isTrue();
+        assertThat(actual).isEqualTo(expected);
+    }
```

Comment:
아래 테스트 코드는 통과해야할것 같아보입니다. 그러나 지금 상태로는 통과하지 않는 것 같습니다.

```suggestion
    @DisplayName("질문 데이터는 삭제 후 soft delete 처리가 된다.")
    @Test
    void set_delete_true_when_answer_was_deleted() {
        DeleteHistory expected = new DeleteHistory(ContentType.ANSWER,
                                                   A1.getId(),
                                                   A1.getWriter(),
                                                   LocalDateTime.now());

        NsUser writer = A1.getWriter();
        NsUser loginUser = new NsUser(writer.getId(), writer.getUserId(), writer.getPassword(), writer.getName(), writer.getEmail());

        DeleteHistory actual = answer.delete(loginUser);

        assertThat(answer.isDeleted()).isTrue();
        assertThat(actual).isEqualTo(expected);
    }
```